core/src/main/java/org/apache/iceberg/hadoop/HadoopTableOperations.java

```diff
@@ -82,6 +95,104 @@ public TableMetadata current() {
     return currentMetadata;
   }
 
+  @Override
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  public EncryptionManager encryption() {
+    // TODO run by single thread? or synchronize?
+    if (null != encryptionManager) {
+      return encryptionManager;
+    }
+
+    TableMetadata tableMetadata = current();
+    Map<String, String> tableProperties = tableMetadata.properties();
+
+    String keyManagerType = PropertyUtil.propertyAsString(tableProperties,
+            TableProperties.ENCRYPTION_MANAGER_TYPE, TableProperties.ENCRYPTION_MANAGER_TYPE_PLAINTEXT);
+    if (keyManagerType.equals(TableProperties.ENCRYPTION_MANAGER_TYPE_PLAINTEXT)) {
+      encryptionManager = new PlaintextEncryptionManager();
+      return encryptionManager;
+    }
+
+    if (keyManagerType.equals(TableProperties.ENCRYPTION_MANAGER_TYPE_SINGLE_ENVELOPE) ||
+            keyManagerType.equals(TableProperties.ENCRYPTION_MANAGER_TYPE_DOUBLE_ENVELOPE)) {
+      boolean doubleWrap = keyManagerType.equals(TableProperties.ENCRYPTION_MANAGER_TYPE_DOUBLE_ENVELOPE);
+      if (doubleWrap) {
+        throw new RuntimeException("Double envelope encryption is not supported yet");
+      }
```

Comment:
We may just remove these logic since double envelope isn't supported any way, and at the end of the method we throw exception for any unsupported encryption type.

Reply:
agreed